examples/sse_chat.rs

```diff
@@ -111,19 +105,16 @@ fn user_connected(
 
     // When `drx` will dropped then `dtx` will be canceled.
     // We can track it to make sure when the user leaves chat.
-    warp::spawn(poll_fn(move || dtx.poll_cancel()).map(move |_| {
+    warp::spawn(async move {
+        dtx.closed().await;
+        drx.close();
```

Comment:
That usage hold the `oneshot::channel` open forever, so the `fn user_disconnected(...)` not reachable.